internal/mq/msgstream/mq_msgstream.go

```diff
@@ -185,11 +183,12 @@ func (ms *mqMsgStream) Start() {
 }
 
 func (ms *mqMsgStream) Close() {
-
 	ms.streamCancel()
-	ms.readerLock.Lock()
+	if len(ms.receiveBuf) > 0 {
```

Comment:
better to write like this:
select {
      case _= <- ms.receiveBuf:
     default:
}